packages/logger/src/logger/ogma.ts

```diff
@@ -114,102 +183,149 @@ export class Ogma {
     return colorize(levelString, color, this.styler, this.options.color);
   }
 
-  private wrapInBrackets(valueToBeWrapper: string): string {
-    return `[${valueToBeWrapper}]`;
+  private wrapInBrackets(valueToBeWrapped: string): string {
+    let retVal = this.wrappedValues[valueToBeWrapped];
+    if (retVal) {
+      return retVal;
+    }
+    retVal = `[${valueToBeWrapped}]`;
+    this.wrappedValues[valueToBeWrapped] = retVal;
+    return retVal;
   }
 
   private formatJSON(
     message: any,
-    { application = '', correlationId = '', context = '', level, ...meta }: PrintMessageOptions,
+    level: LogLevel,
+    {
+      application = this.application,
+      correlationId,
+      context = this.options.context,
+      ...meta
+    }: OgmaPrintOptions,
   ): string {
-    let json: Partial<OgmaLog> = {
-      time: this.getTimestamp(),
-    };
-    delete meta.formattedLevel;
-    const mappedLevel = this.options.levelMap[LogLevel[level] as keyof typeof LogLevel];
+    const mappedLevel = this.asString(
+      this.options.levelMap[LogLevel[level] as keyof typeof LogLevel],
+    );
 
-    if (this.options.logHostname) {
-      json.hostname = this.hostname;
-    }
+    let fastJson = `{"time":${Date.now()},`;
 
-    if (this.options.logApplication) {
-      json.application = application || this.options.application || undefined;
-    }
+    fastJson += `"hostname":${this.asString(this.hostname)},`;
+    fastJson += `"pid":${this.jsonPid},`;
+    fastJson += `"ool":${this.asString(LogLevel[level])},`;
+    fastJson += `"level":${mappedLevel},`;
 
-    if (this.options.logPid) {
-      json.pid = Number(this.pid);
-    }
+    if (application) fastJson += `"application":${this.asString(application)},`;
+
+    if (correlationId) fastJson += `"correlationId":${this.asString(correlationId)},`;
+
+    if (context) fastJson += `"context":${this.asString(context)},`;
+
+    if (meta && Object.keys(meta).length > 0) fastJson += `"meta":${stringify(meta)},`;
+
+    if (this.options.levelKey) fastJson += `"${this.options.levelKey}":${mappedLevel},`;
+
+    if (typeof message === 'object')
+      fastJson += `"message":${stringify(message, this.circularReplacer())}`;
```

Comment:
I think it will have the same behavior as `JSON.stringify`, for some primitives it will handle but for things like `Symbol` or `BigInt`, it brokes or return `undefined`.

Reply:
My thought here is that we already handle the cases of `Symbol`, `Function`, and `BigInt` in the circular replacer, so could we just use `fastJson += "message":${stringify(message, this.circularReplacer()}`; and not worry about the if check here? Not sure it would change the performance, but it's worth the though